app/models/indexing/index_subqueue.rb

```diff
@@ -120,8 +120,12 @@ def add_document_to_batch(obj)
   end
 
   def add_deletion_to_batch(id)
-    basics = { "_index" => klass.index_name, "_type" => klass.document_type, "_id" => id }
-    @batch << { delete: basics }.to_json
+    if klass.respond_to? :index_name 
+      basics = { "_index" => klass.index_name, "_type" => klass.document_type, "_id" => id }
+      @batch << { delete: basics }.to_json
+    else
+      Rails.logger.error "Brownshirt: #{klass.class.name} passed to add_deletion_to_batch Unknown"
```

Comment:
This isn't related the the functionality, but could you change this to another color shirt, or even "Browncoat," like in Firefly? "Brownshirt" usually refers to a Nazi, and I think it would be better to avoid including the term in our codebase, even though it's not an intentional reference.